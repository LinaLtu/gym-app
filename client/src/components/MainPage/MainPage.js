/* eslint-disable react-hooks/rules-of-hooks */
import React, { useState } from 'react';
import moment from 'moment';
import { useQuery } from 'react-query';
import Header from '../Header';
import SearchField from '../SearchField';
import CoursesList from '../CoursesList';
import Paginator from '../Paginator';
import { getCourses } from '../../services/courseService'

import './MainPage.scss';

const CSS_NAME = 'main-page'

const MainPage = () => {
    const [categories, setCategories] = useState([]);
    const [date, setDate] = useState(moment());
    const [page, setPage] = useState(1);

    let dataLength;

    let { data } = useQuery(['courses', page, date, categories], (_, page, date, categories) => getCourses(moment(date), categories, page));

    if (!data) {
        data = [];
    }

    dataLength = data.length;

    // Instead of react-query, the useEffect() hook could be used

    // useEffect(() => {
    //     getCourses(moment(date), categories, page)
    //         .then(responseData => setData(responseData))
    // }, [page, date, categories])


    return (
        <div className={CSS_NAME}>
            <Header />
            <SearchField
                categories={categories}
                page={page}
                setCategories={setCategories}
                setDate={setDate}
            />
            <CoursesList courses={data} />
            <Paginator currentPage={page} setPage={setPage} dataLength={dataLength} />
        </div>
    );
}

export default MainPage;
