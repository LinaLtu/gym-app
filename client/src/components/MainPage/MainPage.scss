html, body {
    max-width: 100%;
    overflow-x: hidden;
}

.main-page {
    position: relative;

    &_error {
        position: absolute;
        top: 40%;
        left: 40%;
        margin-right: -50%;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        width: 200px;
        height: 100px;
        margin-top: 200px;
        border: 2px solid red;
        border-radius: 5px;
        margin: 50px;
        box-shadow: 0 0 6px 0 rgba(0,0,0,0.2);
        margin: auto 0;
    }
}

