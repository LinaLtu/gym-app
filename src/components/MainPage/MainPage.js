import React from 'react';
import Header from '../Header';
import SearchField from '../SearchField';
import CoursesList from '../CoursesList';

import './MainPage.scss';

const CSS_NAME = 'main-page'

const MainPage = () => {
    return (
        <div className={CSS_NAME}>
            <Header />
            <SearchField />
            <CoursesList />
        </div>
    );
}

export default MainPage;