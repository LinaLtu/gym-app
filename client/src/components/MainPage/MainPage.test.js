import React from 'react';
import renderer from 'react-test-renderer';

import MainPage from './';

jest.mock('../Header', () => 'Header');
jest.mock('../SearchField', () => 'SearchField');
jest.mock('../CoursesList', () => 'CoursesList');
jest.mock('../Paginator', () => 'Paginator');


describe('MainPage Component', () => {
    it('renders according to design', () => {
        const component = renderer.create(
            <MainPage />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});