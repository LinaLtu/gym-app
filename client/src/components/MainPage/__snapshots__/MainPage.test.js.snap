// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders according to design 1`] = `
<div
  className="main-page"
>
  <Header />
  <SearchField
    date={"2020-08-06T12:28:22.086Z"}
    setCategories={[Function]}
    setDate={[Function]}
  />
  <CoursesList
    courses={Array []}
  />
  <Paginator
    currentPage={1}
    dataLength={0}
    setPage={[Function]}
  />
</div>
`;
